### DISCUSSION TITLE
Saving machine code binaries

### DISCUSSION THREAD

**Discussion by gdm413229** [jkotlinski/durexforth#380] [Q&A] [answered]
Is there a way to compile your masterpiece written in Forth into machine code and save the resulting machine code to disk or cassette? Distributing a Forth system along with the masterpiece's source code and with the data files required by said masterpiece will be wasteful in terms of disk space, especially in a 1541-formatted disk (or .d64 disk image, they both hold up to 664 blocks (of 256 bytes each!) of file data).

**Comment by Whammo** [comment, marked as answer]
Durexforth offers utilities to save your program in a turn-key fashion by including
the turnkey module once the program is ready to be saved.

2.7.3 Turn-key Operation

**Reply by gdm413229** [reply]
How is a Forth program made independent from the interpreter?

**Reply by Whammo** [reply]
You redirect the start vector by:
```forth
' myprogram start !

```
 before you save-prg.
If your program is a closed loop, it won't drop back to the interpreter

**Reply by gdm413229** [reply]
Does the durexForth compiler take Forth code and emit 6502 machine language out the other end?

**Reply by Whammo** [reply]
The code section contains pure 6502 machine code.

**Reply by gdm413229** [reply]
Can you give me a breakdown of the start vector redirection line?

**Reply by gdm413229** [reply]
Does the save-prg word in the turnkey lexicon take the machine code in the code section and save them to disk? Do you need to pass a special command to the BASIC interpreter just to run binaries saved by the save-prg word in durexForth?

**Reply by Whammo** [reply]
' (tick) finds the execution token, or perhaps the start address of the word that follows it. In this case, "myprogram" and puts that address on the top of the stack.
START, finds the address of the startup word vector (a value) and puts its address on the top of the stack.
! (store) saves the word under the top of the stack to the address on the top of the stack. And, then removes both from the stack.

**Reply by Whammo** [reply]
For the most part, Durexforth programs require a configurable amount of the run-time to execute.
  So yes, it saves the machine code in the code section.

 Just load "myprogram",8 and run
